addons/controllers/clusterbootstrap_controller.go

```diff
@@ -320,6 +327,25 @@ func (r *ClusterBootstrapReconciler) createOrPatchPackageInstallSecret(cluster *
 		}
 	}
 
+	// Add cluster and package labels to secrets if not already present
+	// This helps us to track the secrets in the watch and trigger Reconcile requests when
+	// these secrets are updated
+	updateLabels := false
+	if secret.Labels == nil {
+		secret.Labels = map[string]string{}
+		updateLabels = true
+	} else if secret.Labels[addontypes.PackageNameLabel] != pkg.RefName ||
+		secret.Labels[addontypes.ClusterNameLabel] != cluster.Name {
+		updateLabels = true
+	}
+	if updateLabels {
+		secret.Labels[addontypes.PackageNameLabel] = pkg.RefName
+		secret.Labels[addontypes.ClusterNameLabel] = cluster.Name
+		r.Update(r.context, secret)
```

Comment:
question: 1) directly modifying the pointer is risky, do we want to have a deep copy ? 2) do we want to handle update error ? It is possible we hit an update error when the obj has been modified by other controller/process. 3) this logic only adds labels to local object, do we want to apply the same labels to the secrets which will be created on remote workload cluster ?

Reply:
Fixed

Reply:
For this "do we want to apply the same labels to the secrets which will be created on remote workload cluster", would like to understand more about the usecase here